cmd/docker/docker.go

```diff
@@ -28,16 +28,43 @@ import (
 	"go.opentelemetry.io/otel"
 )
 
+var errCtxUserTerminated = errors.New("user terminated the process")
+
 func main() {
-	err := dockerMain(context.Background())
-	if err != nil && !errdefs.IsCancelled(err) {
+	ctx := context.Background()
+	err := dockerMain(ctx)
+	if err != nil && !errdefs.IsCancelled(err) && !errors.Is(err, errCtxUserTerminated) {
```

Comment:
Hmm, I'm wondering. In which case would `err` be `IsCancelled` and not caused by L54?

Reply:
When we cancel a prompt it returns an `IsCancelled` error.

Reply:
But isn't the prompt cancellation always a direct consequence of SIGINT?

Reply:
yes, but it was wired up to replace the error from `context.Cancelled` to `errdefs.Cancelled`
https://github.com/docker/cli/blob/master/cli/command/utils.go#L114-L117
https://github.com/docker/cli/blob/master/cli/command/utils.go#L78

Reply:
So why not just:
```suggestion
	if err != nil && !errdefs.IsCancelled(err) && !errors.Is(err, context.Canceled) {
```

?

Reply:
specifying an exact error here at least allows us to not accidentally ignore all context cancellation errors and instead target only those the user cancelled

Reply:
In which case the `context.Canceled` error isn't caused by user interrupt?

Reply:
In this PR we _try_ to target user interrupt errors as a custom error using `context.Cause`. For the Go std library (like `http.Transport`) it would use `context.Cause` which means it returns the custom error we are injecting instead of `context.Canceled`.

This PR might not cover all cases where a user interrupt is handled with `context.Cause`. But it is a start!